training/training_population.py

```diff
@@ -0,0 +1,49 @@
+from utils import text_extractor
+import validators
+from unittest import TestCase
+
+SIZE_OF_TRAINING_EXAMPLE = 5
+
+
+class WebsiteIterator:
+    def __init__(self, url):
+        self.url = url
+        self.content = self.get_website_text()
+        self.index = 0
+        self.content_length = len(self.content)
+
+    def get_website_text(self):
+        all_text = ''
+        if self.is_website_url(self.url):
+            all_text, title = text_extractor.get_all_text_from_url(self.url)
+        else:
+            all_text = self.url
+        return all_text
+
```

Comment:
this is a workaround, usually, we want a constructor to accept only one type.
this approach is confusing, I will accept it for now, may open a ticket to user "Factory Design Pattern" (welcome to read about)
also you could been avoid it if you will learn on Mocks - for next time.